x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -718,16 +735,21 @@ public int forceEvict(Predicate<KeyType> cacheKeyPredicate) {
         });
         var evictedCount = 0;
         if (matchingEntries.isEmpty() == false) {
+            long nonZeroEvictedCount = 0;
             synchronized (this) {
                 for (Entry<CacheFileRegion> entry : matchingEntries) {
                     boolean evicted = entry.chunk.forceEvict();
+                    if (evicted && entry.freq > 0) {
+                        nonZeroEvictedCount++;
+                    }
                     if (evicted && entry.chunk.io != null) {
                         unlink(entry);
                         keyMapping.remove(entry.chunk.regionKey, entry);
                         evictedCount++;
                     }
```

Comment:
Why not:
```
if (evicted && entry.chunk.io != null) {
    if (evicted && entry.freq > 0) {
        nonZeroEvictedCount++;
    }
    unlink(entry);
    keyMapping.remove(entry.chunk.regionKey, entry);
    evictedCount++;
}
```
I'm less familiar with the internals of the cache (@original-brownbear help me out here), but seeing as there's already an `evictedCount`, wouldn't we want to increment in the same place?  Is it actually considered evicting if `entry.chunk.io == null`?

Reply:
>  Is it actually considered evicting if entry.chunk.io == null

Not really I'd say but it doesn't matter much. That should only happen in a corner case anyway (so many active pages being written to that we evict before we find a free IO instance for one) that should be impossible for practical cache settings.